code/processes/dqe.q

```diff
@@ -1,34 +1,25 @@
 \d .dqe
 
-/- default parameters
-subscribeto:@[value;`subscribeto;`];                          /-list of tables subscribed to
-subscribesyms:@[value;`subscribesyms;`];                      /-list of syms subscribed to
-schema:@[value;`schema;0b];                                   /-retrieve schema from tickerplant
-tickerplanttypes:@[value;`tickerplanttypes;`tickerplant];     /-list of tickerplant types to try and make a connection to
-replaylog:@[value;`replaylog;0b];                             /-replay the tickerplant log file
-tpconnsleepintv:@[value;`tpconnsleepintv;10];                 /-default wait time before trying to reconnect to discovery/TP
-/- end of default parameters
-
-subscribe:{[]
-  if[count s:.sub.getsubscriptionhandles[tickerplanttypes;();()!()];
-  .lg.o[`subscribe;"found available tickerplant, attempting to subscribe"];
-  /-set the date that was returned by the subscription code i.e. the date for the tickerplant log file
-  /-and a list of the tables that the process is now subscribing for
-  subinfo: .sub.subscribe[subscribeto;subscribesyms;schema;replaylog;first s];
-  /-setting subtables and tplogdate globals
-  .dqe,:subinfo]
-  }
-
 init:{
-  .servers.CONNECTIONS:distinct .servers.CONNECTIONS, .dqe.tickerplanttypes;
   .lg.o[`init;"searching for servers"];
-  .servers.startup[];
-  .servers.startupdependent[tickerplanttypes;tpconnsleepintv];
-  subscribe[];
+  .servers.startup[];                                                                                           /- Open connection to discovery
   }
 
-tableExists:{x in tables[]};                                  /-function to check for table, param is table name as a symbol
+tableexists:{x in tables[]};                                                                                    /- function to check for table, param is table name as a symbol
+
+runcheck:{[fn;vars;rs]                                                                                          /- function used to send other function to test processes
+  if[0<count s:{x where 0=x in raze exec procname,proctype from .servers.querydiscovery`ALL}[rs];
+    .lg.e[`process;string s," processes are not connectable"]];
```

Comment:
Should we return early at this point? Seems like it this error msg gets output there's no point continuing with function

Reply:
Thought it would be good to still get back results of processes that where entered correctly or would it be better to maybe set up an email alert to say that a config file has wrong input?

Reply:
I don't follow... are the situations where you get no results & print an error here, but _do_ get handles in the next line?

Reply:
If an input process does not exist (or can't get connected too), a null will be returned when checking with discovery. So then no handle will be opened for it. As long as one of the inputs is correct, the function still runs and will return results for the correct processes. 
This was a problem when looking at the results as there was nothing to say that anything went wrong. We added this so that the incorrect inputs are returned so that the user knows that these are the ones that did not run.

Reply:
> If an input process does not exist (or can't get connected too), a null will be returned when checking with discovery. So then no handle will be opened for it.

Yeah, so both this & the next line will fail? I'm still not really getting it, it seems to me like there are two checks for the same thing? Can you give me an example of something that would output the error on the first check, but _not_ error on the second?

Reply:
Oh I think I see, this would be if you passed in e.g. `` `rdb1`rdb2`` but you only had `` `rdb1`` running, is that right?